src/peer_config.rs

```diff
@@ -25,15 +26,21 @@ pub const DEFAULT_IDLE_TIMEOUT_MSEC: u64 = 30_000; // 30secs
 pub const DEFAULT_KEEP_ALIVE_INTERVAL_MSEC: u32 = 10_000; // 10secs
 
 pub fn new_client_cfg(peer_cert_der: &[u8]) -> R<quinn::ClientConfig> {
-    let peer_cert = quinn::Certificate::from_der(peer_cert_der)?;
+    // FIXME: Seems quinn have `ParseError` as a return type of a public interface but the type
+    // itself is private. Hence using this workaround to collect it via `format!`. Ideally should
+    // be just convertible inside quick_error as usual with other errors.
+    let peer_cert = quinn::Certificate::from_der(peer_cert_der)
+        .map_err(|e| Error::CertificateParseError(format!("{:?}, {}", e, e)))?;
 
     let mut peer_cfg_builder = {
         let mut client_cfg = quinn::ClientConfig::default();
         client_cfg.transport = Arc::new(new_transport_cfg(None, None));
 
         quinn::ClientConfigBuilder::new(client_cfg)
     };
-    let _ = peer_cfg_builder.add_certificate_authority(peer_cert)?;
+    let _ = peer_cfg_builder
```

Comment:
We should handle this result.

Reply:
This one should be handled: it goes like `let _ = peer_cfg_builder.map_err(..)?`, so the discarded result here is only `Ok(ConfigBuilder)` which we have no use for.